src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -487,15 +494,29 @@ private int zipAlign(Path buildToolsPath, String unalignedApk, String alignedApk
     }
 
     private int sign(Path buildToolsPath, String alignedApk) throws IOException, InterruptedException {
-        createDevelopKeystore();
+        ReleaseConfiguration releaseConfiguration = projectConfiguration.getReleaseConfiguration();
+        String keyStorePath = releaseConfiguration.getProvidedKeyStorePath();
+        String keyStorePass = releaseConfiguration.getProvidedKeyStorePassword();
+        String keyAlias = releaseConfiguration.getProvidedKeyAlias();
+        String keyPass = releaseConfiguration.getProvidedKeyAliasPassword();
+        if (keyStorePath == null ||
+                !keyStorePath.endsWith(".keystore") || !Files.exists(Path.of(keyStorePath)) ||
+                keyAlias == null || keyStorePass == null || keyPass == null) {
+            // default to debug keystore
+            keyStorePath = createDevelopKeystore().toString();
+            keyStorePass = DEFAULT_DEBUG_KEY_STORE_PASSWORD;
+            keyAlias = DEFAULT_DEBUG_KEY_ALIAS;
+            keyPass = DEFAULT_DEBUG_KEY_ALIAS_PASSWORD;
+        }
 
         String apkSignerCmd = buildToolsPath.resolve("apksigner").toString();
         ProcessRunner sign =  new ProcessRunner(apkSignerCmd, "sign", "--ks",
-                Constants.USER_SUBSTRATE_PATH.resolve(Constants.ANDROID_KEYSTORE).toString(),
-                "--ks-key-alias", "androiddebugkey",
-                "--ks-pass", "pass:android",
-                "--key-pass", "pass:android",
-                alignedApk);
+                keyStorePath, "--ks-key-alias", keyAlias);
+        sign.addArg("--ks-pass");
+        sign.addSecretArg("pass:" + keyPass);
```

Comment:
Are you sure that `keyPass` and `keyStorePass` shouldn't be switched?

Reply:
oh, good catch. Fixed